tests/v2prov/tests/machineprovisioning/machine_test.go

```diff
@@ -649,3 +651,171 @@ func Test_Provisioning_MP_DrainNoDelete(t *testing.T) {
+				MachinePools: []provisioningv1api.RKEMachinePool{{
+					EtcdRole:         true,
+					ControlPlaneRole: true,
+					WorkerRole:       true,
+					Quantity:         &defaults.One,
+				}},
+			},
+		},
+	}
+
+	c, err := cluster.New(clients, base)
+	require.NoError(t, err)
+
+	c, err = cluster.WaitForCreate(clients, c)
+	require.NoError(t, err)
+
+	// Validate that exactly one Machine and capture its template hash
+	machines, err := cluster.Machines(clients, c)
+	require.NoError(t, err)
+	require.Equal(t, 1, len(machines.Items), "expected exactly one machine initially")
+
+	old := machines.Items[0]
+	oldHash := old.Labels["machine-template-hash"]
+	require.NotEmpty(t, oldHash, "old machine missing template-hash label")
+
+	// Create a fresh machine config that can be mutated.
+	newCfgRef, err := nodeconfig.NewPodConfig(clients, c.Namespace)
+	require.NoError(t, err)
+
+	gvrPodCfg := schema.GroupVersionResource{
+		Group: "rke-machine-config.cattle.io", Version: "v1", Resource: "podconfigs",
+	}
+	pc, err := clients.Dynamic.Resource(gvrPodCfg).Namespace(c.Namespace).Get(ctx, newCfgRef.Name, metav1.GetOptions{})
+	require.NoError(t, err)
+
+	ud, ok := unstructuredString(pc.Object, "userdata")
+	require.True(t, ok)
+
+	// Force a no-op template diff
+	pc.Object["userdata"] = ud + `# Noop Change`
+
+	_, err = clients.Dynamic.Resource(gvrPodCfg).Namespace(c.Namespace).Update(ctx, pc, metav1.UpdateOptions{})
+	require.NoError(t, err)
+
+	cur, err := clients.Provisioning.Cluster().Get(c.Namespace, c.Name, metav1.GetOptions{})
+	require.NoError(t, err)
+
+	require.NotNil(t, cur.Spec.RKEConfig)
+	require.GreaterOrEqual(t, len(cur.Spec.RKEConfig.MachinePools), 1)
+
+	// Point the pool at the new PodConfig.
+	cur.Spec.RKEConfig.MachinePools[0].NodeConfig = newCfgRef
+	_, err = clients.Provisioning.Cluster().Update(cur)
+	require.NoError(t, err)
+
+	require.Eventually(t, func() bool {
+		machines, _ = cluster.Machines(clients, c)
+		return len(machines.Items) == 2
+	}, 4*time.Minute, 2*time.Second, "never saw 2 nodes after config change")
+
+	var newName string
+	require.Eventually(t, func() bool {
+		machines, _ = cluster.Machines(clients, c)
+		for _, m := range machines.Items {
+			hash := m.Labels["machine-template-hash"]
+			if hash != "" && hash != oldHash && m.Status.NodeRef != nil {
+				newName = m.Name
+				return true
+			}
+		}
+		return false
+	}, 10*time.Minute, 2*time.Second, "no node with new template hash (and NodeRef) appeared")
+
+	// Ensure the new node reaches Ready=True
+	var newNode *corev1.Node
+	clusterClients, err := clients.ForCluster(c.Namespace, c.Name)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	require.Eventually(t, func() bool {
+		m, err := clients.CAPI.Machine().Get(c.Namespace, newName, metav1.GetOptions{})
+		if err != nil || m.Status.NodeRef == nil {
+			return false
+		}
+
+		n, err := clusterClients.Core.Node().Get(m.Status.NodeRef.Name, metav1.GetOptions{})
+		if err != nil {
+			return false
+		}
+		newNode = n
```

Comment:
```suggestion
		node, err = clusterClients.Core.Node().Get(m.Status.NodeRef.Name, metav1.GetOptions{})
		if err != nil {
			return false
		}
```

I'd just assign directly here.